csm_web/scheduler/admin.py

```diff
@@ -49,273 +96,311 @@ def has_delete_permission(self, request, obj=None):
         return request.user.is_superuser
 
 
+@admin.register(Student)
+class StudentAdmin(CoordAdmin):
+    list_filter = ("active",)
+    list_display = ("name", "get_email", "get_course", "get_mentor", "section")
+    search_fields = ("user__email", "user__first_name", "user__last_name")
+    actions = ("drop_students", "undrop_students")
+    # autocomplete_fields = ("section", "user")
+
+    def get_fields(self, request, obj):
+        fields = ["name", "get_email", "get_course", "section", "get_attendances"]
+        fields.insert(4, "user" if request.user.is_superuser else "get_user")
+        return tuple(fields)
+
+    def get_readonly_fields(self, request, obj):
+        fields = ["get_course", "get_email", "name", "get_attendances"]
+        if not request.user.is_superuser:
+            fields.insert(0, "get_user")
+        return tuple(fields)
+
+    def has_delete_permission(self, request, obj=None):
+        return (
+            False
+        )  # delete sections + mentor by deleting sections, drop by deactivating
+
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        if not request.user.is_superuser:
+            courses = get_visible_courses(request.user)
+            if db_field.name == "section":
+                kwargs["queryset"] = Section.objects.filter(course__in=courses)
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
+    def get_queryset(self, request):
+        qs = super().get_queryset(request)
+        if request.user.is_superuser:
+            return qs
+        return qs.filter(section__course__in=get_visible_courses(request.user))
+
+    def drop_students(self, request, queryset):
+        queryset.update(active=False)
+
+    drop_students.short_description = (
+        "Drop student(s) from a section. This action is reversible, but it is still strongly "
+        "recommended that you double check the name/email/section/course first."
+    )
+
+    def undrop_students(self, request, queryset):
```

Comment:
I think it's important to make clear that this action is the exact inverse of "drop" - "reenroll" might carry with it the implication that it would let you enroll a student in a different section